ksqldb-engine/src/main/java/io/confluent/ksql/physical/pull/HARouting.java

```diff
@@ -200,7 +202,7 @@ private void executeRounds(
     while (processedPartitions < totalPartitions) {
       final Future<PartitionFetchResult> future = completionService.take();
       try {
-        final PartitionFetchResult fetchResult = future.get();
+        final PartitionFetchResult fetchResult = future.get(10, TimeUnit.SECONDS);
```

Comment:
should we make this configurable per-query in case people have really large table scans?

Reply:
Yes we should, thanks for pointing out